### PR TITLE
Add CSI external components images

### PR DESCRIPTION
**Pull request by c3y1huang**
Component includes:
* attacher
* registrar
* provisioner
* resizer
* snapshotter
* livenessprobe

Ref: 
* https://github.com/SUSE/avant-garde/issues/1779
* [Test result](https://github.com/SUSE/caasp-test-cases/pull/78#issuecomment-668415485)

Signed-off-by: Chin-Ya Huang <chin-ya.huang@suse.com>

### CODE REVIEW

**Comment by cclhsu** [comment]
@c3y1huang Does all the container daemon need to run as root? or they can also run as nobody?

**Comment by jenting** [comment]
Generally, LGTM, just the rpm package naming, could the rpm package named with prefix `csi-` to make it easier to understand? Thanks.

**Comment by c3y1huang** [comment]
> @c3y1huang Does all the container daemon need to run as root? or they can also run as nobody?

Upstream are running all containers as root.

**Comment by c3y1huang** [comment]
> Generally, LGTM, just the rpm package naming, could the rpm package named with prefix `csi-` to make it easier to understand? Thanks. Like:
> 
> ```
> csi-external-attacher
> csi-node-driver-registrar
> csi-external-provisioner
> csi-external-resizer
> csi-external-snapshotter
> csi-livenessprobe
> ```

Good suggestion.

File: caasp-livenessprobe-image/caasp-livenessprobe-image.kiwi [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!-- OBS-AddTag: caasp/v4.5/livenessprobe:%%PKG_VERSION%%-rev<VERSION> caasp/v4.5/livenessprobe:%%PKG_VERSION%%-rev<VERSION>-build<RELEASE> caasp/v4.5/livenessprobe:beta -->
+
+<image schemaversion="6.9" name="caasp-livenessprobe-image" xmlns:suse_label_helper="com.suse.label_helper">
```
Comment (jenting) [inline caasp-livenessprobe-image/caasp-livenessprobe-image.kiwi:5]: Could this package name renamed to `caasp-csi-livenessprobe-image` as well, thanks.

**Comment by innobead** [comment]
> > Generally, LGTM, just the rpm package naming, could the rpm package named with prefix `csi-` to make it easier to understand? Thanks. Like:
> > ```
> > csi-external-attacher
> > csi-node-driver-registrar
> > csi-external-provisioner
> > csi-external-resizer
> > csi-external-snapshotter
> > csi-livenessprobe
> > ```
> 
> Good suggestion, but I don't know how to revoke since the RPM SRs has been accepted. Or is it ok if we leave it without `csi-` ?
> 
> WDYT @cclhsu , @innobead ?

I believe definitely having a way to revoke, like resubmit then request to delete old ones? @cclhsu could you help with this? Thanks.

**Comment by innobead** [comment]
LGTM, it will good to rename packages with `csi` prefix. (after updating, then we can approve here)

Good job :+1:

**Comment by cclhsu** [comment]
> > > Generally, LGTM, just the rpm package naming, could the rpm package named with prefix `csi-` to make it easier to understand? Thanks. Like:
> > > ```
> > > csi-external-attacher
> > > csi-node-driver-registrar
> > > csi-external-provisioner
> > > csi-external-resizer
> > > csi-external-snapshotter
> > > csi-livenessprobe
> > > ```
> > 
> > 
> > Good suggestion, but I don't know how to revoke since the RPM SRs has been accepted. Or is it ok if we leave it without `csi-` ?
> > WDYT @cclhsu , @innobead ?
> 
> I believe definitely having a way to revoke, like resubmit then request to delete old ones? @cclhsu could you help with this? Thanks.

Yes, we can send in delete SR. Though I don't have permission for accepted for CaaSP 4.5/5 currently.

**Comment by innobead** [comment]
> > > > Generally, LGTM, just the rpm package naming, could the rpm package named with prefix `csi-` to make it easier to understand? Thanks. Like:
> > > > ```
> > > > csi-external-attacher
> > > > csi-node-driver-registrar
> > > > csi-external-provisioner
> > > > csi-external-resizer
> > > > csi-external-snapshotter
> > > > csi-livenessprobe
> > > > ```
> > > 
> > > 
> > > Good suggestion, but I don't know how to revoke since the RPM SRs has been accepted. Or is it ok if we leave it without `csi-` ?
> > > WDYT @cclhsu , @innobead ?
> > 
> > 
> > I believe definitely having a way to revoke, like resubmit then request to delete old ones? @cclhsu could you help with this? Thanks.
> 
> Yes, we can send in delete SR. Though I don't have permission for accepted for CaaSP 4.5/5 currently.

Understood, please check with @jordimassaguerpla :bow:

File: caasp-livenessprobe-image/caasp-livenessprobe-image.kiwi (hunk 1 above)
Comment (c3y1huang) [reply caasp-livenessprobe-image/caasp-livenessprobe-image.kiwi]: done

File: caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!-- OBS-AddTag: caasp/v4.5/livenessprobe:%%PKG_VERSION%%-rev<VERSION> caasp/v4.5/livenessprobe:%%PKG_VERSION%%-rev<VERSION>-build<RELEASE> caasp/v4.5/livenessprobe:beta -->
```
Comment (jenting) [inline caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi:3]:
```suggestion
<!-- OBS-AddTag: caasp/v4.5/csi-livenessprobe:%%PKG_VERSION%%-rev<VERSION> caasp/v4.5/csi-livenessprobe:%%PKG_VERSION%%-rev<VERSION>-build<RELEASE> caasp/v4.5/csi-livenessprobe:beta -->
```

File: caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+<image schemaversion="6.9" name="caasp-csi-livenessprobe-image" xmlns:suse_label_helper="com.suse.label_helper">
+  <description type="system">
+    <author>SUSE Containers Team</author>
+    <contact>containers@suse.com</contact>
+    <specification>livenessprobe running on a SLES15 SP2 container guest</specification>
```
Comment (jenting) [inline caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi:9]:
```suggestion
    <specification>csi-livenessprobe running on a SLES15 SP2 container guest</specification>
```

File: caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  </description>
+  <preferences>
+    <type
+      image="docker"
+      derived_from="obsrepositories:/suse/sle15#15.2">
+      <containerconfig
+        name="caasp/v4.5/livenessprobe"
```
Comment (jenting) [inline caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi:16]:
```suggestion
        name="caasp/v4.5/csi-livenessprobe"
```

File: caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        tag="%%PKG_VERSION%%"
+        maintainer="SUSE Containers Team &lt;containers@suse.com&gt;">
+        <entrypoint execute="/usr/bin/livenessprobe"/>
```
Comment (jenting) [inline caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi:19]: I'm not sure will you change the binary name to `csi-livenessprobe`, leave it to you :smile:

File: caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        <subcommand clear="true"/>
+        <labels>
+          <suse_label_helper:add_prefix prefix="com.suse.caasp.v4.5">
+            <label name="org.opencontainers.image.description" value="livenessprobe running on a SLES15 SP2 container guest"/>
+            <label name="org.opencontainers.image.title" value="CaaSP livenessprobe container"/>
```
Comment (jenting) [inline caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi:24]: <label name="org.opencontainers.image.title" value="CaaSP csi-livenessprobe container"/>

File: caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+            <label name="org.opencontainers.image.created" value="%BUILDTIME%"/>
+            <label name="org.opencontainers.image.vendor" value="SUSE LLC"/>
+            <label name="org.opencontainers.image.url" value="https://www.suse.com/products/caas-platform/"/>
+            <label name="org.opencontainers.image.version" value="%%PKG_VERSION%%"/>
+            <label name="org.openbuildservice.disturl" value="%DISTURL%"/>
+            <label name="com.suse.reference" value="registry.suse.com/caasp/v4.5/livenessprobe:%%PKG_VERSION%%"/>
```
Comment (jenting) [inline caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi:30]:
```suggestion
            <label name="com.suse.reference" value="registry.suse.com/caasp/v4.5/csi-livenessprobe:%%PKG_VERSION%%"/>
```

File: caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi (hunk 5 above)
Comment (c3y1huang) [reply caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi:19]: I've left the binary as it is. Should be ok.

File: caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi (hunk 2 above)
Comment (c3y1huang) [reply caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi]: done

File: caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi (hunk 3 above)
Comment (c3y1huang) [reply caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi]: done

File: caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi (hunk 4 above)
Comment (c3y1huang) [reply caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi]: done

File: caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi (hunk 6 above)
Comment (c3y1huang) [reply caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi]: done

File: caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi (hunk 7 above)
Comment (c3y1huang) [reply caasp-csi-livenessprobe-image/caasp-csi-livenessprobe-image.kiwi]: done

**Pull request merged**